### ISSUE TITLE
Author's name not rendering

### ISSUE THREAD

**Issue by Benneee** [mdn/express-locallibrary-tutorial#80]
Hello!

The author's name on the book list page does not render because the field being accessed, name, does not exist. The fields in the DB are first_name and family_name.

I will like to submit a PR, but there is no 'Contributing.md' file, how do I contribute to this project?

**Comment by hamishwillee** [comment]
Name is created as a virtual here https://github.com/mdn/express-locallibrary-tutorial/blob/master/models/author.js#L15

**Comment by Benneee** [comment]
There has to be an update in the tutorial about this bit then. 

I forked this repository and ran the project ordinarily before changing anything, of course, after the necessary installations. However, on running the project, an error is found on 'all books' page.
![Screenshot from 2019-11-21 10-25-27](https://user-images.githubusercontent.com/22181511/69327822-72ba0f00-0c4e-11ea-8083-0788df992bf4.png)

**Comment by hamishwillee** [comment]
Thanks, just to be pedantic, you used the package.json for the project (so exactly same environment)?

**Comment by hamishwillee** [comment]
(or to put it another way "I'm really busy on other things at the moment so I want you to have done something wrong, rather than me needing to fix this code" :-)

**Comment by Benneee** [comment]
LOL! You're quite funny. This is hilarious. :)
I can be very pedantic too, so I made sure I wasn't doing anything wrong. 
Same exact package.json, I made sure of that. I basically forked the repository and did the necessary thing.
I know you're busy and I commend your prompt responses, so let's do it like this:
If I do fix this and submit a PR, would you look at it and merge once you are alright with it?

**Comment by hamishwillee** [comment]
That would be brilliant. Thank you!

**Comment by Benneee** [comment]
The virtual is working quite alright. I think it has to do with the data inside the author object coming from the bookController, I used `(#{book.author})` and it worked, showing me the author object for each book right there on the view, however, on trying to access either the first_name or family_name properties or even using the virtual, I get a 500/Undefined error like what the snapshot above shows.

I'll investigate more and submit a PR once I'm successful.
Do have a great day!

**Comment by Spikatrix** [comment]
Can repro.

Seems to be caused by some books that doesn't have an author. Here's the output when I change `#{book.author.name}` to `#{book.author}`:

```
Apes and Angels ({ _id: 5dd65897ed6fcbfe5fffbc12, first_name: 'Ben', family_name: 'Bova', date_of_birth: 1932-11-07T16:00:00.000Z, __v: 0 })
Apes and Angels ({ _id: 5dd65947a15dd6fe6354d6e4, first_name: 'Ben', family_name: 'Bova', date_of_birth: 1932-11-07T16:00:00.000Z, __v: 0 })
BBw ({ _id: 5dcd00278e6c734a34c73530, first_name: 'test', family_name: 'ok', date_of_birth: 2019-11-09T00:00:00.000Z, date_of_death: 2019-11-21T00:00:00.000Z, __v: 0 })
bs ({ _id: 5dccfd032251cc3eb870d8bc, first_name: 'Ami', family_name: 'aa', date_of_birth: 2019-11-13T00:00:00.000Z, date_of_death: 2019-11-14T00:00:00.000Z, __v: 0 })
Cant teach an old dog new tricks. yes you can ({ _id: 5dc562e2f75ac82e88c5fe7e, first_name: 'Hellp', family_name: 'Hellp', date_of_birth: 1999-01-01T00:00:00.000Z, date_of_death: 2220-01-01T00:00:00.000Z, __v: 0 })
Death Wave ({ _id: 5dd65897ed6fcbfe5fffbc12, first_name: 'Ben', family_name: 'Bova', date_of_birth: 1932-11-07T16:00:00.000Z, __v: 0 })
Death Wave ({ _id: 5dd65947a15dd6fe6354d6e4, first_name: 'Ben', family_name: 'Bova', date_of_birth: 1932-11-07T16:00:00.000Z, __v: 0 })
End of the world ({ _id: 5dd10ee0e29e0c0a54786f4d, first_name: 'dog', family_name: 'dog', date_of_birth: 0001-01-01T00:00:00.000Z, date_of_death: 0001-01-03T00:00:00.000Z, __v: 0 })
hello ({ _id: 5dce803b7fd00c25d85239d3, first_name: 'azhjjjj', family_name: 'dfnfsgns', date_of_birth: 1900-01-01T00:00:00.000Z, date_of_death: 1984-10-10T00:00:00.000Z, __v: 0 })
Ibuliini ({ _id: 5dd836299282a93a64ab52b5, first_name: 'Miguel', family_name: 'Sundstrom', date_of_birth: 1960-01-01T00:00:00.000Z, date_of_death: null, __v: 0 })
Laurens Book ()
Lorem ({ _id: 5dd61f9e6f5cd9064b67d714, first_name: 'qweqw', family_name: 'qweqwe', date_of_birth: 2019-02-12T00:00:00.000Z, date_of_death: null, __v: 0 })
mi vida ({ _id: 5dcaedf1383b4d2ae0a27d22, first_name: 'Perico', family_name: 'palotes', date_of_birth: 1998-12-25T00:00:00.000Z, date_of_death: null, __v: 0 })
pavelonäge ({ _id: 5ddf9a99335a5b1e44fc1e43, first_name: 'pavle', family_name: 'adnsad', date_of_birth: 1312-12-03T00:00:00.000Z, date_of_death: 1991-12-31T00:00:00.000Z, __v: 0 })
Test ({ _id: 5dce803b7fd00c25d85239d3, first_name: 'azhjjjj', family_name: 'dfnfsgns', date_of_birth: 1900-01-01T00:00:00.000Z, date_of_death: 1984-10-10T00:00:00.000Z, __v: 0 })
Test Book 1 ({ _id: 5dd65948a15dd6fe6354d6e7, first_name: 'Jim', family_name: 'Jones', date_of_birth: 1971-12-16T00:00:00.000Z, __v: 0 })
Test Book 1 ({ _id: 5dd65897ed6fcbfe5fffbc15, first_name: 'Jim', family_name: 'Jones', date_of_birth: 1971-12-16T00:00:00.000Z, __v: 0 })
Test Book 2 ({ _id: 5dd65897ed6fcbfe5fffbc15, first_name: 'Jim', family_name: 'Jones', date_of_birth: 1971-12-16T00:00:00.000Z, __v: 0 })
Test Book 2 ({ _id: 5dd65948a15dd6fe6354d6e7, first_name: 'Jim', family_name: 'Jones', date_of_birth: 1971-12-16T00:00:00.000Z, __v: 0 })
Teste1 ({ _id: 5dccfd032251cc3eb870d8bc, first_name: 'Ami', family_name: 'aa', date_of_birth: 2019-11-13T00:00:00.000Z, date_of_death: 2019-11-14T00:00:00.000Z, __v: 0 })
The last book of Boby Bob ()
The Name of the Wind (The Kingkiller Chronicle, #1) ({ _id: 5dd65945a15dd6fe6354d6e3, first_name: 'Patrick', family_name: 'Rothfuss', date_of_birth: 1973-06-06T00:00:00.000Z, __v: 0 })
The Name of the Wind (The Kingkiller Chronicle, #1) ({ _id: 5dd65894ed6fcbfe5fffbc11, first_name: 'Patrick', family_name: 'Rothfuss', date_of_birth: 1973-06-06T00:00:00.000Z, __v: 0 })
The Slow Regard of Silent Things (Kingkiller Chronicle) ({ _id: 5dd65945a15dd6fe6354d6e3, first_name: 'Patrick', family_name: 'Rothfuss', date_of_birth: 1973-06-06T00:00:00.000Z, __v: 0 })
The Slow Regard of Silent Things (Kingkiller Chronicle) ({ _id: 5dd65894ed6fcbfe5fffbc11, first_name: 'Patrick', family_name: 'Rothfuss', date_of_birth: 1973-06-06T00:00:00.000Z, __v: 0 })
The Wise Man's Fear (The Kingkiller Chronicle, #2) ({ _id: 5dd65945a15dd6fe6354d6e3, first_name: 'Patrick', family_name: 'Rothfuss', date_of_birth: 1973-06-06T00:00:00.000Z, __v: 0 })
The Wise Man's Fear (The Kingkiller Chronicle, #2) ({ _id: 5dd65894ed6fcbfe5fffbc11, first_name: 'Patrick', family_name: 'Rothfuss', date_of_birth: 1973-06-06T00:00:00.000Z, __v: 0 })
tunes ()
Un buen titulo ({ _id: 5dccfd032251cc3eb870d8bc, first_name: 'Ami', family_name: 'aa', date_of_birth: 2019-11-13T00:00:00.000Z, date_of_death: 2019-11-14T00:00:00.000Z, __v: 0 })
```

Could it be that someone is messing around with the DB?

**Comment by Benneee** [comment]
Yeah, this is exactly what I explained above. The issue that some books do not have authors should be checked though. I'll come back in a bit..... Thanks

Well done Spikatrix!

**Comment by Benneee** [comment]
OK, I think the DB needs to be checked.

I tried checking for the two values first in the virtual before returning the value, that didn't work either.

**Comment by hamishwillee** [comment]
> Could it be that someone is messing around with the DB?

Very likely. Easiest way to check is to use a clean db. I can clear the default one if you like.

**Comment by Benneee** [comment]
By all means, please clear the DB. Thank you very much.

**Comment by hamishwillee** [comment]
Done! There was a heap of crap in there.

**Comment by Benneee** [comment]
Hi Hamish...

Finally submitted that PR. I populated the DB and wrote a check for the name virtual. 
For the change you see in the package.lock.json, this was added after I ran "npm install" after cloning the project.
Here's the [link to the PR](https://github.com/mdn/express-locallibrary-tutorial/pull/83)

Thank you so very much for the good work you're doing and your time.

**Comment by jsingh-github** [comment]
Im getting the same issue. How do i fix it please

**Comment by hamishwillee** [comment]
Closing - @Benneee very kindly fixed this back in December in #83. Let's pretend I was a lot more prompt :-0
